pkg/ucp/frontend/controller/resourceproviders/getresourceprovidersummary.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package resourceproviders
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	http "net/http"
+	"strings"
+
+	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
+	armrpc_controller "github.com/radius-project/radius/pkg/armrpc/frontend/controller"
+	armrpc_rest "github.com/radius-project/radius/pkg/armrpc/rest"
+	"github.com/radius-project/radius/pkg/middleware"
+	"github.com/radius-project/radius/pkg/ucp/datamodel"
+	"github.com/radius-project/radius/pkg/ucp/datamodel/converter"
+	"github.com/radius-project/radius/pkg/ucp/resources"
+	"github.com/radius-project/radius/pkg/ucp/store"
+)
+
+var _ armrpc_controller.Controller = (*GetResourceProviderSummary)(nil)
+
+// GetResourceProviderSummary is the controller implementation to get the list of resources stored in a resource group.
+type GetResourceProviderSummary struct {
+	armrpc_controller.Operation[*datamodel.ResourceProviderSummary, datamodel.ResourceProviderSummary]
+}
+
+// NewGetResourceProviderSummary creates a new controller for listing resources stored in a resource group.
+func NewGetResourceProviderSummary(opts armrpc_controller.Options) (armrpc_controller.Controller, error) {
+	return &GetResourceProviderSummary{
+		Operation: armrpc_controller.NewOperation(opts,
+			armrpc_controller.ResourceOptions[datamodel.ResourceProviderSummary]{
+				RequestConverter:  converter.ResourceProviderSummaryDataModelFromVersioned,
+				ResponseConverter: converter.ResourceProviderSummaryDataModelToVersioned,
+			},
+		),
+	}, nil
+}
+
+// Run implements controller.Controller.
+func (r *GetResourceProviderSummary) Run(ctx context.Context, w http.ResponseWriter, req *http.Request) (armrpc_rest.Response, error) {
+	relativePath := middleware.GetRelativePath(r.Options().PathBase, req.URL.Path)
+
+	scope, name, err := r.extractScopeAndName(relativePath)
```

Comment:
This is a special case where the request URL isn't a resource ID. This is why I had to write bespoke controllers for the summary.